src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -336,6 +336,19 @@ def create_virtual_machine(resource_group_name, vm_params, network_interfaces, a
         }
       }
 
+      capacity_reservation_group_id = vm_params[:capacity_reservation_group]
+      unless capacity_reservation_group_id.nil?
+        unless vm_params[:capacity_reservation_group].start_with?('/subscriptions')
+          capacity_reservation_group_id = rest_api_url(REST_API_PROVIDER_COMPUTE, 'capacityReservationGroups', resource_group_name: resource_group_name, name: vm_params[:capacity_reservation_group])
```

Comment:
Is there a good reason to enable two ways of passing this value to the CPI?

Could we drop this `unless` if the documentation / validation is clear that a simple (no `/subscriptions/...` prefix) value is required?

Reply:
Hey @aramprice, thanks for the review! The idea behind this `unless` is that capacity reservation groups could be managed from a different resource group than the one used for your bosh deployment. I can't say if people would want to do that, but in my specific use case, it is actually part of the same resource group. Therefore, I don't mind removing it.

Reply:
I have removed the support for fully qualified resource IDs in my latest commit. If anyone in the community needs this feature, we can easily reintroduce it. It might be better to have a separate configuration parameter in that case.